[doc] Venice Push Job Diagram

KaiSernLim opened the pull request:
## Summary 

Adding Venice Push Job Diagram and a diagram for various entities in the write path to the docs.

There is a link to the source diagram on Whimsical, and you'll need to copy the diagram to your account in order to edit it. Also included an `.svg` file copy from Whimsical's experimental feature, but it's kinda glitchy and missing some lines.

<img width="4502" height="3259" alt="VenicePushJob@2x" src="https://github.com/user-attachments/assets/bdcefd28-a550-4ab8-9e84-5654efdcb0c0" />

<img width="4801" height="2867" alt="WritePathQuantities@2x" src="https://github.com/user-attachments/assets/cc2dee5d-1513-4f9a-be3b-1a51a53bc7bb" />


## Does this PR introduce any user-facing changes?
<!--
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, choose 'No'.
-->
- [X] No. You can skip the rest of this section.

ZacAttack reviewed (commented):
Looks great!  A few comments:

nix 'glorified pipe' on VPJ
Who is the intended audience for this?  Developers or new users of Venice?  The START_VALIDATE_SCHEMA_AND_BUILD_DICT_MAP_JOB is a developer level of information, but if this is meant for users of Venice then it's probably TMI.

'Same key could be mentioned multiple times' this.... 'can' happen on a user input, but the job by default is supposed to fail when this is detected because we can't make guarantees on the result otherwise from the users perspective.

Dali is mentioned as an input format, buuuuuut Dali isn't open sourced.  Though we do talk about Dali in external blog posts.  So idk if that changes weather or not you'd like to include it here.

sushantmane reviewed (approved):
Thanks a ton! Let's address @ZacAttack  comments! 🙏 🚀

KaiSernLim commented:
> Looks great! A few comments:
> 
> nix 'glorified pipe' on VPJ Who is the intended audience for this? Developers or new users of Venice? The START_VALIDATE_SCHEMA_AND_BUILD_DICT_MAP_JOB is a developer level of information, but if this is meant for users of Venice then it's probably TMI.
> 
> 'Same key could be mentioned multiple times' this.... 'can' happen on a user input, but the job by default is supposed to fail when this is detected because we can't make guarantees on the result otherwise from the users perspective.
> 
> Dali is mentioned as an input format, buuuuuut Dali isn't open sourced. Though we do talk about Dali in external blog posts. So idk if that changes weather or not you'd like to include it here.

Thanks for the review @ZacAttack! Great to see you! 🚀 

1. Removed the `'glorified pipe'`.
1. Developers are more the intended audience. Today, some team members were discussing that the diagram should be put up on the docs, which I agreed with. I do think that some users may find it helpful to have a mental model about how it all works, if they're stuck on a particular part and trying to understand what is going on.
1. Removed the `'Same key could be mentioned multiple times'` and statement about Dali.

sushantmane reviewed (approved):
Thanks!
